### ISSUE TITLE
Requesting all properties to be returned during search results in `AttributeError`

### ISSUE THREAD

**Issue by nmandery** [stac-utils/stac-fastapi#338]
Requesting all properties to be returned for the search results using the [syntax described in the spec](https://github.com/radiantearth/stac-api-spec/tree/master/fragments/fields) ends either in this exception or the properties being omitted:

```
INFO:     10.104.105.105:4806 - "GET /search?collections=dem30&limit=1000&bbox=35.21,-17.78,35.48,-17.59&fields=properties HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.9/site-packages/brotli_asgi/__init__.py", line 76, in __call__
    await responder(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/gzip.py", line 35, in __call__
    await self.app(scope, receive, self.send_with_gzip)
  File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 219, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 154, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.9/site-packages/starlette/concurrency.py", line 40, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/stac_fastapi/api/routes.py", line 75, in _endpoint
    func(request=request, **request_data.kwargs()), response_class
  File "/usr/local/lib/python3.9/site-packages/stac_fastapi/sqlalchemy/core.py", line 242, in get_search
    resp = self.post_search(search_request, request=kwargs["request"])
  File "/usr/local/lib/python3.9/site-packages/stac_fastapi/sqlalchemy/core.py", line 439, in post_search
    filter_kwargs = search_request.fields.filter_fields
  File "/usr/local/lib/python3.9/site-packages/stac_fastapi/extensions/core/fields/request.py", line 56, in filter_fields
    "include": self._get_field_dict(include),
  File "/usr/local/lib/python3.9/site-packages/stac_fastapi/extensions/core/fields/request.py", line 37, in _get_field_dict
    field_dict[parent].add(key)
AttributeError: 'ellipsis' object has no attribute 'add'
```

The reason for this is a unsound creation of the `field_dict` in `PostFieldsExtension._get_field_dict` in combination with `properties.datetime` being part of the `default_includes`. So once the field `properties` is requested, there are two possibilities depending on the order of the items in the `fields` variable:

* If `properties` comes before `properties.datetime` the value will be set to `...` and the following call to the expected `Set.add` method will fail with the above exception.
* If `properties.datetime` comes before `datetime` no exception occurs, but neither will the requested properties be part of the response.

I will submit a small PR to adress this.

**Comment by geospatial-jeff** [comment]
Closed with #339.  Thanks @nmandery !

**Comment by nmandery** [comment]
Thank you for merging @geospatial-jeff
